lib/data_explorer_query_group_bookmarkable.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DataExplorerQueryGroupBookmarkable < BaseBookmarkable
+  def self.model
+    DataExplorer::QueryGroup
+  end
+
+  def self.serializer
+    UserDataExplorerQueryGroupBookmarkSerializer
+  end
+
+  def self.preload_associations
+    [:data_explorer_queries, :groups]
+  end
+
+  def self.list_query(user, guardian)
+    group_ids = user.groups.pluck(:id) 
+    return if group_ids.empty?
+    user.bookmarks_of_type("DataExplorer::QueryGroup")
+    .joins("INNER JOIN data_explorer_query_groups ON data_explorer_query_groups.id = bookmarks.bookmarkable_id
+          AND bookmarks.bookmarkable_type = 'DataExplorer::QueryGroup'")
+      .joins("LEFT JOIN data_explorer_queries ON data_explorer_queries.id = data_explorer_query_groups.query_id")
+      .joins("LEFT JOIN groups ON groups.id = data_explorer_query_groups.group_id")
```

Comment:
I'm wondering if the two `LEFT JOIN` here are necessary since we don't seem to rely on the joins for further filtering.

Reply:
Yeah agree with TGX on the join and `bookmarks_of_type` comments -- if we need to do further filtering later on we can always add these back in.

Reply:
Oh right of course, thanks for that.